### PR TITLE
Pokemons

### PR DESCRIPTION
**Pull request by ki3kia**

### CODE REVIEW

File: src/PokemonList.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  results: Array<Pokemon>;
+}
+
+export const PokemonList = (): JSX.Element => {
+  const section = useRef<HTMLElement>(null);
+  const limit = 10;
+
+  const [fetchRes, setFetchRes] = useState<PokemonResponse>({
+    next: `https://pokeapi.co/api/v2/pokemon/?offset=0&limit=${limit}`,
+    results: [],
```
Comment (TheForsakenSpirit) [inline src/PokemonList.tsx:19]: try to fetch new page without previous data

**Review by TheForsakenSpirit — CHANGES_REQUESTED** [review]

File: src/App.css [hunk 2]
```diff
@@ -1,38 +1,38 @@
 .App {
-  text-align: center;
+    text-align: center;
```
Comment (mrlika) [inline src/App.css:2]: All the CSS files should be formatted with Prettier

**Review by mrlika — CHANGES_REQUESTED** [review]

File: src/PokemonList.tsx [hunk 3]
```diff
@@ -3,13 +3,27 @@ import React, { useEffect, useRef, useState } from 'react';
...
+  if (!arg && typeof arg !== 'object') return false;
+  return 'next' in (arg as PokemonResponse) && 'results' in (arg as PokemonResponse);
+}
+
+function isPokemon(arg: unknown): arg is Pokemon {
+  if (!arg && typeof arg !== 'object') return false;
+  if ('name' in (arg as Pokemon) && 'url' in (arg as Pokemon)) {
+    return true;
+  }
+  return false;
```
Comment (mrlika) [inline src/PokemonList.tsx:24]: Simplify
```suggestion
  return 'name' in (arg as Pokemon) && 'url' in (arg as Pokemon);
```

File: src/PokemonList.tsx [hunk 4]
```diff
@@ -46,15 +68,24 @@ export const PokemonList = (): JSX.Element => {
...
   return (
-    <section id='pokeList' ref={section}>
-      {fetchRes.results.map((pokemon) => (
-        <div key={pokemon.name}>{pokemon.name}</div>
+    <section id='pokeList' className='pokemonList' ref={section}>
+      {fetchRes.results.map((pokemon, key) => (
+        <div key={pokemon.name}>
+          <img
+            src={`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${
+              key + 1
```
Comment (mrlika) [inline src/PokemonList.tsx:76]: `pokemon.id` should be used here. not index.

**Review by mrlika — CHANGES_REQUESTED** [review]

**Review by TheForsakenSpirit — CHANGES_REQUESTED** [review]

File: src/PokemonList.tsx [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+}
+
+function isPokemon(arg: unknown): arg is Pokemon {
+  if (!arg && typeof arg !== 'object') return false;
+  return 'name' in (arg as Pokemon) && 'url' in (arg as Pokemon);
+}
+
+export const PokemonList = (): JSX.Element => {
+  const section = useRef<HTMLElement>(null);
+  const PAGE_SIZE = 10;
```
Comment (TheForsakenSpirit) [inline src/PokemonList.tsx:26]:
```suggestion
  
const PAGE_SIZE = 10;

export const PokemonList = (): JSX.Element => {
  const section = useRef<HTMLElement>(null);
```

**Review by TheForsakenSpirit — CHANGES_REQUESTED** [review]

File: src/PokemonList.tsx [hunk 6]
```diff
@@ -44,6 +45,14 @@ export const PokemonList = (): JSX.Element => {
...
 
         const responseList = await getPokemons(url);
+        if (isPokemonRes(responseList) && responseList) {
+          for (let i = 0; i < responseList.results.length; i++) {
+            const responsePokemon = await getPokemons(responseList.results[i].url);
+            responseList.results[i] = isPokemon(responsePokemon)
+              ? responsePokemon
+              : { ...responseList.results[i], id: 0 };
+          }
+        }
```
Comment (TheForsakenSpirit) [inline src/PokemonList.tsx:55]: Convert to `map` cycle.

**Review by TheForsakenSpirit — CHANGES_REQUESTED** [review]

**Review by TheForsakenSpirit — APPROVED** [review]

**Review by mrlika — APPROVED** [review]

**Pull request merged**